File: AgentRL/controller/internal/types/controller.go
package types

type WorkerData struct {
	Id        int    `json:"id"`
	Address   string `json:"address"`
	Capacity  int    `json:"capacity"`
	Current   int    `json:"current"`
	LastVisit string `json:"last_visit"`
	Status    string `json:"status"`
	Stale     bool   `json:"stale"`
	Locked    bool   `json:"locked"`
}

type TaskData struct {
	Indices []TaskIndex        `json:"indices"`
	Workers map[int]WorkerData `json:"workers"`
}

type ListWorkersResponse map[string]TaskData

type SessionData struct {
	Name       string      `json:"name"`
	Index      TaskIndex   `json:"index"`
	CustomTask interface{} `json:"custom_task"`
	Start      string      `json:"start"`
	LastUpdate string      `json:"last_update"`
	WorkerId   int         `json:"worker_id"`
	Locked     bool        `json:"locked"`
	Cancelling bool        `json:"cancelling"`
}

type ListSessionsResponse map[int]SessionData

File: AgentRL/controller/dashboard/src/types/controller.ts
export interface WorkerData {
  id: number
  address: string
  capacity: number
  current: number
  last_visit: string  // '%Y-%m-%d %H:%M:%S' at UTC+8
  status: string      // 'ALIVE' | 'COMA' | 'DEAD'
  stale?: boolean
  locked: boolean
}

export type TaskIndex = string | number

export interface TaskData {
  indices: TaskIndex[]
  workers: Record<number, WorkerData>
}

export interface SessionData {
  name: string
  index: TaskIndex
  custom_task?: any
  start: string        // '%Y-%m-%d %H:%M:%S' at UTC+8
  last_update: string  // '%Y-%m-%d %H:%M:%S' at UTC+8
  worker_id: number
  locked: boolean
  cancelling: boolean
}

export type ListWorkersResponse = Record<string, TaskData>

export type ListSessionsResponse = Record<number, SessionData>

export interface GetVersionResponse {
  variant: string
  version: number
}
